CocoIDE/cocol.py

```diff
@@ -240,14 +257,18 @@ def deploy(name, addr):
 
 
 def link(
-    objectfiles=[], ideobjtext=None, filename="linkout", termp=False, fileout=True
-):
+    objectfiles: list = [],
+    ideobjtext=None,
+    filename: str = "linkout",
+    termp: bool = False,
+    fileout=True,  # Fileout is never used here and the type of ideobjtext is ambiguous
+):  # objectfiles shouldn't be mutable, instead the list should be passed (default=None)
```

Comment:
It isn't mutated?

Reply:
I mean it would work as is, but really mutable items shouldn't be set as default values in function parameters. What should happen is it is set to None or something and then the empty list is passed in.

Reply:
…but `objectfiles` isn't mutated